lib/events/azsessions/azsessions.go

```diff
@@ -320,7 +320,7 @@ func (h *Handler) CompleteUpload(ctx context.Context, upload events.StreamUpload
 	// cleaned up before a new attempt
 
 	parts = slices.Clone(parts)
-	slices.SortFunc(parts, func(a, b events.StreamPart) bool { return a.Number < b.Number })
+	slices.SortFunc(parts, func(a, b events.StreamPart) int { return int(a.Number - b.Number) })
```

Comment:
Could we avoid the subtraction?

```suggestion
	slices.SortFunc(parts, func(a, b events.StreamPart) int { return cmp.Compare(a.Number, b.Number) })
```

Same for others.

Reply:
OK, fair enough. Since then I've learned to avoid subtraction for comparisons, as it can cause hard-to-detect overflow issues.